pychromecast/dial.py

```diff
@@ -90,7 +90,8 @@ def _get_status(
             _LOGGER.debug("Resolved service %s to %s", service, host)
             break
 
-    headers = {"content-type": "application/json"}
+    # unsetting the host header, as requests with a domain would be blocked otherwise
+    headers = {"host": "", "content-type": "application/json"}
```

Comment:
I think this end up setting it to an empty string. I am not sure that is such a good idea.

"A Host header field must be sent in all HTTP/1.1 request messages. A [400 Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) status code may be sent to any HTTP/1.1 request message that lacks or contains more than one Host header field."

So it likely need to be set to the IP address of the request.

Reply:
I would still like an example showing where this is a problem and how this PR fixes it.

Reply:
I suspect just setting a manual dns (ie not mdns) hostname for a chromecast instead of an IP would cause it.

Reply:
Just realized that it might only be possible with the changes for ipv6 that was posted that does a getaddrinfo lookup.

Reply:
My usecase is indeed to have a custom DNS name for a cast device, as I can't use mDNS in my setup, since the cast devices are in a different L2/L3 network than Home Assistant.

I do agree with setting the `Host` header to the resolved IP, that's a cleaner way to solve this.